jax/_src/lax/parallel.py

```diff
@@ -1269,7 +1284,8 @@ def _axis_index_bind(*, axis_name):
     nreps = dynamic_axis_env.nreps
     trace = frame.pmap_trace
 
-    out_aval = ShapedArray((), np.int32)
+    out_aval = _axis_index_abstract_eval(
```

Comment:
Do we actually need to update this non-omnistaging code? I'm curious if tests fail without it.